### PR TITLE
[hailgenetics] update the hailgenetics images, fix Makefile

### PR DESCRIPTION
**Pull request by danking**
Now these images live at gcr.io/hail-vdc/hailgenetics/hail and gcr.io/hail-vdc/hailgenetics/genetics.

### CODE REVIEW

File: docker/Makefile [hunk 1]
```diff
@@ -108,3 +83,31 @@ deploy: push
...
+HAIL_IMAGE = hailgenetics/hail
+GENETICS_IMAGE = hailgenetics/genetics
+
+.PHONY: publish-hail-public-image
+publish-hail-public-image: hail-ubuntu-stmp
+	make -C ../hail python/hail/hail_pip_version
+	cp ../hail/python/hail/hail_pip_version hail/hail_pip_version
+	-docker pull $(GCR_PREFIX)/$(HAIL_IMAGE)
+	python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"hail-ubuntu"}}' Dockerfile.base Dockerfile.base.out
+	docker build hail/ -f hail/Dockerfile.out -t $(HAIL_IMAGE):$$(cat hail/hail_pip_version) \
```
Comment (jigold) [inline docker/Makefile:99]: Can we use more variables here? You repeat `$$(cat hail/hail_pip_version)` a lot and it makes it harder to read. Maybe also a variable for `$(HAIL_IMAGE):$$(cat hail/hail_pip_version)`

File: docker/Makefile [hunk 2]
```diff
@@ -108,3 +83,31 @@ deploy: push
...
+     --cache-from $(GCR_PREFIX)/$(HAIL_IMAGE),$(HAIL_IMAGE)
+	docker push $(HAIL_IMAGE):$$(cat hail/hail_pip_version)
+	docker tag $(HAIL_IMAGE):$$(cat hail/hail_pip_version) $(GCR_PREFIX)/$(HAIL_IMAGE):$$(cat hail/hail_pip_version)
+	docker push $(GCR_PREFIX)/$(HAIL_IMAGE):$$(cat hail/hail_pip_version)
+
+.PHONY: publish-genetics-public-image
+publish-genetics-public-image: publish-hail-public-image
+	python3 ../ci/jinja2_render.py '{"hail_public_image":{"image":"'$(HAIL_IMAGE):$$(cat hail/hail_pip_version)'"}}' genetics/Dockerfile genetics/Dockerfile.out
+	-docker pull $(GCR_PREFIX)/$(GENETICS_IMAGE)
+	docker build genetics/ -f genetics/Dockerfile.out -t $(GENETICS_IMAGE):$$(cat hail/hail_pip_version) \
```
Comment (jigold) [inline docker/Makefile:109]: Same comment as above.

File: batch/batch/worker/worker.py [hunk 3]
```diff
@@ -261,8 +261,7 @@ def __init__(self, job, name, spec):
             tag = 'latest'
 
         if repository in HAIL_GENETICS_IMAGES:
-            repository_name_without_prefix = repository[len(HAIL_GENETICS):]
-            repository = f'gcr.io/{PROJECT}/{repository_name_without_prefix}'
+            repository = f'gcr.io/{PROJECT}/{repository}'
```
Comment (jigold) [inline batch/batch/worker/worker.py:264]: The batch pool executor tests will catch if this is wrong -- correct?

File: docker/genetics/Dockerfile [hunk 4]
```diff
@@ -5,6 +5,9 @@ RUN hail-apt-get-install \
     libbz2-dev \
     libcurl4-openssl-dev \
     liblzma-dev \
+    libvcflib-tools \
```
Comment (jigold) [inline docker/genetics/Dockerfile:8]: Why did you add these?

File: docker/genetics/Dockerfile [hunk 5]
```diff
@@ -16,8 +19,6 @@ RUN mkdir samtools && \
      ./configure --without-curses && \
      make && \
      make install)
-RUN git clone --recursive https://github.com/vcflib/vcflib.git && \
```
Comment (jigold) [inline docker/genetics/Dockerfile:19]: Why did you delete this line?

**Review by jigold — CHANGES_REQUESTED** [review]

File: docker/genetics/Dockerfile (hunk 4 above)
Comment (danking) [reply docker/genetics/Dockerfile:8]: Compiling VCFlib was hard and they now have an apt-get package

File: docker/Makefile (hunk 1 above)
Comment (danking) [reply docker/Makefile]: It's difficult to do what you're looking for in Make, so I rewrote this as a shell script. Let me know what you think.

File: batch/batch/worker/worker.py (hunk 3 above)
Comment (danking) [reply batch/batch/worker/worker.py:264]: Yeah, the images would be absent and that would cause an error. I've already uploaded the images to GCR using the first version of this code. I just ran it again now to ensure the new thing is right too.

File: docker/genetics/Dockerfile (hunk 5 above)
Comment (danking) [reply docker/genetics/Dockerfile:19]: compiling VCFlib was hard

File: docker/publish-public-images.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+           -t $docker \
+           -t $google \
+           -t $latest \
+           --cache-from $latest,$base
+    docker push $docker
+    docker push $google
+    docker push $latest
+}
+
+python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"hail-ubuntu"}}' Dockerfile.base Dockerfile.base.out
```
Comment (jigold) [inline docker/publish-public-images.sh:32]: Does this actually work? It looks like it's expecting the Dockerfile to be located at `hail/Dockerfile.out`

File: docker/publish-public-images.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+make -C ../hail python/hail/hail_pip_version
+cp ../hail/python/hail/hail_pip_version hail/hail_pip_version
+hail_pip_version=$(cat hail/hail_pip_version)
+
+build_and_push() {
+    name=$1
+    base=$2
+
+    docker=hailgenetics/$name:$hail_pip_version
```
Comment (jigold) [inline docker/publish-public-images.sh:15]: Did you decide not to tag these with the SHA anymore?

**Review by jigold — DISMISSED** [review]

File: docker/publish-public-images.sh (hunk 6 above)
Comment (danking) [reply docker/publish-public-images.sh:32]: ah nope, I should've cleaned before I tested.

File: docker/publish-public-images.sh (hunk 7 above)
Comment (danking) [reply docker/publish-public-images.sh:15]: These images were never tagged with a SHA, they're tagged with the pip version because their version of hail is from PyPI e.g. https://hub.docker.com/r/hailgenetics/genetics/tags?page=1&ordering=last_updated

Comment (jigold) [reply docker/publish-public-images.sh:15]: Sorry. I should have been more clear. We used to have this as well: `GENETICS_PUBLIC_IMAGE = gcr.io/$(PROJECT)/genetics-public:$(shell docker images -q --no-trunc genetics-public:latest | sed -e 's,[^:]*:,,')`

File: docker/publish-public-images.sh [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+           -t $docker \
+           -t $google \
+           -t $latest \
+           --cache-from $latest,$base
+    docker push $docker
+    docker push $google
+    docker push $latest
+}
+
+python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"hail-ubuntu"}}' hail/Dockerfile.base hail/Dockerfile.base.out
```
Comment (jigold) [inline docker/publish-public-images.sh:32]: Maybe this was a different PR, but I think the Dockerfile name is not correct here.

**Review by jigold — CHANGES_REQUESTED** [review]

**Review by jigold — APPROVED** [review]
